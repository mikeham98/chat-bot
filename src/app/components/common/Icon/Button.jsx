import React from 'react';
import styles from '../../../../themes/icon.scss';
import Icon from "./index";

export default class IconButton extends React.PureComponent {
    render() {
        const {onClick, type, icon, style, color, hoverColor, height, width, disabled} = this.props;
        return (
            <button
                type={type}
                onClick={onClick}
                disabled={disabled}
                style={style}
                className={styles.iconButton}
            >
                <Icon
                    IconComponent={icon}
                    color={color}
                    hoverColor={hoverColor}
                    height={height}
                    width={width}
                />
            </button>
        );
    }
}