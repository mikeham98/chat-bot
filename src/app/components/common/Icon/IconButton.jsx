import React from 'react';
import styles from '../../../../themes/components/icon.scss';
import Icon from "./index";

const IconButton = ({onClick, type, icon, style, color, hoverColor, height, width, disabled}) => {
    return (
        <button
            type={type}
            onClick={onClick}
            disabled={disabled}
            style={style}
            className={styles.iconButton}
        >
            <Icon
                IconComponent={icon}
                color={color}
                hoverColor={hoverColor}
                height={height}
                width={width}
            />
        </button>
    );
};

export default React.memo(IconButton);