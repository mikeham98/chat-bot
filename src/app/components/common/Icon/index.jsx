import React from 'react';

export default class Icon extends React.PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            isHovered: false
        };
        this.handleMouseEnter = this.handleMouseEnter.bind(this);
        this.handleMouseLeave = this.handleMouseLeave.bind(this);
    }

    handleMouseEnter() {
        this.setState({isHovered: true})
    };

    handleMouseLeave() {
        this.setState({isHovered: false})
    };

    render() {
        const {IconComponent, color, hoverColor, height, width} = this.props;
        const {isHovered} = this.state;
        return (
            <IconComponent
                fill={(isHovered && hoverColor) || color}
                height={height}
                width={width}
                onMouseEnter={this.handleMouseEnter}
                onMouseLeave={this.handleMouseLeave}
            />
        );
    }
}

Icon.defaultProps = {
    height: '40px',
    width: '40px'
};