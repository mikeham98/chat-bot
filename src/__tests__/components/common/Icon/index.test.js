import React from 'react';
import {shallow} from 'enzyme';
import Icon from '../../../../app/components/common/Icon/index';

describe('Icon', () => {
    let shallowedWrapper;
    let props = {};
    const wrapper = () => {
        if (!shallowedWrapper) {
            shallowedWrapper = shallow(<Icon {...props}/>);
        }
        return shallowedWrapper;
    };

    beforeEach(() => {
        props = {
            IconComponent: () => <div>IconComponent</div>,
            color: 'red',
            hoverColor: 'blue',
            height: '20px',
            width: '20px'
        };
        shallowedWrapper = undefined;
    });
    describe('render', () => {
        it('should return IconComponent', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });
    describe('functionality', () => {
        describe('fill color', () => {
            it('should be red', () => {
                expect(wrapper().find('IconComponent').first().props().fill).toBe('red');
            });
            it('should change from red to blue (on mouse enter) and back to red (on mouse leave)', () => {
                expect(wrapper().find('IconComponent').first().props().fill).toBe('red');
                wrapper().find('IconComponent').first().props().onMouseEnter();
                expect(wrapper().find('IconComponent').first().props().fill).toBe('blue');
                wrapper().find('IconComponent').first().props().onMouseLeave();
                expect(wrapper().find('IconComponent').first().props().fill).toBe('red');
            });
        });
    });
});

Icon.defaultProps = {
    height: '40px',
    width: '40px'
};