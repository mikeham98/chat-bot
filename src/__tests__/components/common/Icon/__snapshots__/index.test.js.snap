// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon render should return IconComponent 1`] = `
<IconComponent
  fill="red"
  height="20px"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  width="20px"
/>
`;
